atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java

```diff
@@ -1078,6 +1092,233 @@ public void testMarkTableInvolvedSkipsLockChecksOnCommitIfReadDoneDuringTransact
+        sanityCheckOnSortedCells(rows, cells, cellsWrittenOriginally);
+
+        // we write to a cell that has been read to introduce conflict
+        Transaction t2 = startTransaction();
+        Cell cell = cellsWrittenOriginally.get(new Random().nextInt(cellsWrittenOriginally.size()));
+        put(t2, PtBytes.toString(cell.getRowName()), PtBytes.toString(cell.getColumnName()), "v0_0");
+        t2.commit();
+
+        assertThatThrownBy(t1::commit).isInstanceOf(TransactionSerializableConflictException.class);
+    }
+
+
+    @Test
+    public void testGetSortedColumnsReadWriteConflictAtEndOfBatch() {
+        List<byte[]> rows = generateRows(5);
+        List<Cell> cellsWrittenOriginally = generateCells(rows, generateColumns(5));
+        writeCells(cellsWrittenOriginally);
+
+        Transaction t1 = startTransactionWithSerializableConflictChecking();
+        Iterator<Map.Entry<Cell, byte[]>> sortedColumns = t1.getSortedColumns(
+                TEST_TABLE,
+                rows,
+                BatchColumnRangeSelection.create(PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, 100));
+
+        List<Cell> cells = Streams.stream(sortedColumns).map(Map.Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, cells, cellsWrittenOriginally);
+
+        // we write to a cell that has been read to introduce conflict
+        Transaction t2 = startTransaction();
+        Cell cell = cellsWrittenOriginally.get(cellsWrittenOriginally.size() - 1);
+        put(t2, PtBytes.toString(cell.getRowName()), PtBytes.toString(cell.getColumnName()), "v0_0");
+        t2.commit();
+
+        assertThatThrownBy(t1::commit).isInstanceOf(TransactionSerializableConflictException.class);
+    }
+
+    @Test
+    public void testGetSortedColumnsReadWriteConflictForTerminalColumn() {
+        List<byte[]> rows = generateRows(1);
+        byte[] onlyRow = Iterables.getOnlyElement(rows);
+        byte[] lastColumnName = RangeRequests.getLastColumnName();
+        List<Cell> cellsWrittenOriginally = ImmutableList.of(Cell.create(onlyRow, lastColumnName));
+        writeCells(cellsWrittenOriginally);
+
+        Transaction t1 = startTransactionWithSerializableConflictChecking();
+        Iterator<Map.Entry<Cell, byte[]>> sortedColumns = t1.getSortedColumns(
+                TEST_TABLE,
+                rows,
+                BatchColumnRangeSelection.create(PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, 100));
+
+        List<Cell> cells = Streams.stream(sortedColumns).map(Map.Entry::getKey).collect(Collectors.toList());
+        sanityCheckOnSortedCells(rows, cells, cellsWrittenOriginally);
+
+        // we write to a cell that has been read to introduce conflict
+        Transaction t2 = startTransaction();
+        put(t2, onlyRow, lastColumnName, "v0_0");
+        t2.commit();
+
+        assertThatThrownBy(t1::commit).isInstanceOf(TransactionSerializableConflictException.class);
+    }
```

Comment:
Ideally we'd have multiple rows: the reason I asked for this test was mainly to ensure that the case where we stop in the middle of `(r1, c), (r2, c) ...` where `c` _is_ the last column is handled correctly, while this goes through the "finished iterating" case if I'm reading it correctly. Otherwise looks good.